blaze-core/src/main/scala/org/http4s/blazecore/util/Http1Writer.scala

```diff
@@ -28,16 +30,16 @@ import scala.concurrent._
 
 private[http4s] trait Http1Writer[F[_]] extends EntityBodyWriter[F] {
   final def write(headerWriter: StringWriter, body: EntityBody[F]): F[Boolean] =
-    fromFutureNoShift(F.delay(writeHeaders(headerWriter))).attempt.flatMap {
-      case Right(()) =>
-        writeEntityBody(body)
-      case Left(t) =>
-        body.drain.compile.drain.handleError { t2 =>
-          // Don't lose this error when sending the other
-          // TODO implement with cats.effect.Bracket when we have it
-          Http1Writer.logger.error(t2)("Error draining body")
-        } *> F.raiseError(t)
-    }
+    fromFutureNoShift(F.delay(writeHeaders(headerWriter)))
+      .guaranteeCase {
+        case ExitCase.Completed =>
+          F.unit
+
+        case ExitCase.Error(_) | ExitCase.Canceled =>
+          body.drain.compile.drain.handleError { t2 =>
+            Http1Writer.logger.error(t2)("Error draining body")
+          }
+      } >> writeEntityBody(body)
```

Comment:
Did we lose raising the error here?

Reply:
`guaranteeCase` will pass further the raised error from the computing of `FutureNoShift(F.delay(writeHeaders(headerWriter)))` with adding finally-computation `body.drain.compile.drain`. If I'm correctly get it.